### PR TITLE
packages: add roles to all packages

### PR DESCRIPTION
**Pull request by Rugvip**
Part of implementing #8729

This adds the role to all packages, using `yarn backstage-cli migrate packages-roles`, and a few manual tweaks

### CODE REVIEW

File: plugins/search-backend-node/package.json [hunk 1]
```diff
@@ -10,6 +10,9 @@
     "main": "dist/index.cjs.js",
     "types": "dist/index.d.ts"
   },
+  "backstage": {
+    "role": "plugin-backend"
```
Comment (iamEAP) [inline plugins/search-backend-node/package.json:14]: Perhaps `node-library`?  Basically only responsible for wiring up search engine / indexing; doesn't expose routes.

File: packages/search-common/package.json [hunk 2]
```diff
@@ -10,6 +10,9 @@
     "main": "dist/index.cjs.js",
     "types": "dist/index.d.ts"
   },
+  "backstage": {
+    "role": "node-library"
```
Comment (iamEAP) [inline packages/search-common/package.json:14]: This package actually only contains types at the moment, and ends up being included in both frontend and backend.

File: plugins/search-backend-node/package.json (hunk 1 above)
Comment (Rugvip) [reply plugins/search-backend-node/package.json]: Yep, makes sense 👍

File: packages/search-common/package.json (hunk 2 above)
Comment (Rugvip) [reply packages/search-common/package.json]: It's because it doesn't output `esm`, but let's get that fixed instead. I switched it over.

**Review by freben — APPROVED** [review]
Yassss. No template changes?

**Comment by Rugvip** [comment]
It's all experimental for now, I'd even say we may want to hold off merging this one. The rest of the role related PRs aren't on top of this.

A particular thing I'm not sure about is the relation between build and bundle and how they interact with repo-wide commands and what their packages scripts are supposed to be. I think there will be some breaking changes from where this is right now, so don't wanna roll anything out broadly yet

**Comment by Rugvip** [comment]
Alright, this is ready and needs another round of review. I've made quite a few tweaks to the script migration command, and also renamed the plugin roles:

```
plugin-frontend -> frontend-plugin
plugin-frontend-module -> frontend-plugin-module
plugin-backend -> backend-plugin
plugin-backend-module -> backend-plugin-module
```

The old name kept confusing me, because it didn't use the same naming logic as the other roles, like `web-library`. With those it's `<scope>-<function>`, but with the plugins it was the other way around. I think this way around makes them much more intuitive.

The `test` command also got `--passWithNoTests` added as a default, as that either had to be added in places, or worst case the `test` script was being removed.

**Comment by freben** [comment]
I really appreciate the rename. Indeed much cleaner, and reads more fluently/naturally

**Comment by Rugvip** [comment]
Added in another rename of `script` -> `package`. So rather than

```js
"build": "backstage-cli script build"
```

It's now

```js
"build": "backstage-cli package build"
```

Felt that made more sense, seeing as the thing we're actually building, linting or testing is a package.

Also thinking a rename of `repo` -> `project` would make sense, but will leave that for later

**Comment by freben** [comment]
Maybe controversial opinion - can we also get rid of "app" and name that "frontend"? I never liked the "backend" vs "app" pair of words. "frontend" -> "frontend-plugin", etc

**Comment by Rugvip** [comment]
@freben hmm, not sure tbh. I tried out changing it and it does get a bit weird. The main thing being that the package is currently called `app`. There's also other pieces like the pattern of calling the entry point in frontends an "app", e.g. `create-react-app`.

That said a `frontend` role as such I think is fine and fits alright with `backend`. Not actually a huge change to correct that in the future either if we feel that's needed. Gonna go through with the rest of the role renaming, but ofc keep the package name and paths for now.

**Review by benjdlambert — APPROVED** [review]
I APPROVE THIS PR.

**Comment by Rugvip** [comment]
THANKS.

**Pull request merged**